### ISSUE TITLE
[Enhancement] Support a single device with multiple sensors

### ISSUE THREAD

**Issue by librehat** [t0bst4r/home-assistant-matter-hub#110]
![image](https://github.com/user-attachments/assets/807818e9-185e-465b-a413-f1255b033eb8)

Example, my Dyson purifier is now bridged as 3 entities. They show up as 3 separate devices in my SmartThings app. Ideally, this shows up as a single device via Matter that has the capabilities of a switch (for fan control), and humidity and temperature sensors.

**Comment by KipK** [comment]
Also Google Home won't find Humidity on  a separated device if there's alrady a thermostat or a temperature sensor in the same room.
By priority it search first thermostat, then temperature sensor, then other device. If one of those is present it then fail to find the data.

**Comment by KipK** [comment]
> Doesn't seem that we can do anything about it?

Yes, multiple entities on one device. 
If you have a device exposing temp and hum clusters then it works. That's also why I've featured request to have humidity in thermostats 😉

**Comment by t0bst4r** [comment]
Yes, it makes sense for climates, but they also have it in one entity. 

Imaginary Szenario:

> User has a device "my device" in HA with the following entities
> - switch.my_device
> - sensor.my_device_temperature
> - sensor.my_device_humidity
> - light.my_device_light
> - binary_sensor.my_device_open

This device could be anything. Let’s say a door sensor with temp, humidity and light included. I don’t know why, but we don‘t know all possible devices manufacturers could build.

The Problem:
Matter needs a single device type. Devices must only define clusters which are allowed in their "mandatory" or "optional" list. Additionally it does not make sense to add a levelControl to a sensor device to control its light’s brightness. So **one device** to combine all related entities will not be enough.

The solution:
As far as I understood: In matter we‘d need an extra aggregator device (device type: aggregator) which includes multiple nested devices (one per entity). That's totally possible, but quite some change in the code base that I am not sure when to do. And the more I wait, the more users will hate me on release date when they need to re-configure all their rooms in Alexa 😂

**Comment by KipK** [comment]
There's many Matter device on the specs with multiple clusters that would match on HA with multiple entities

from matter-1-3-device-library-specification.pdf

as example, 7.10. Air Quality Sensor

![image](https://github.com/user-attachments/assets/ed0b0a6c-8408-4ca4-8e26-aca800958f97)

Imagine the mess with all those entities separated on your Alexa or GH :)

**Comment by nicoodeimos** [comment]
Just wanted to support this idea too, this would be the best feature ever! Grouping entities into a single « device » could be game changer compared to Home Assistant HomeKit bridge witch sadly doesn’t allow nested accessories (although it is perfectly supported in the protocol and Home app)!

![image](https://github.com/user-attachments/assets/f4212e5e-1a6a-480a-b8f3-58f786579192)

Pleeeeease make this happen!

**Comment by ThatBlockyPenguin** [comment]
I also would love this! I get that it's a lot of work and a big change, but if it's feasible at some point in the future, that's fantastic!

**Comment by ChaotenKurt** [comment]
I agree, it would be great if a device could have multiple entities. For example, if sockets could then directly display the power consumption later on. Or, for example, my Aqara window contact also has a built-in temperature sensor. It would be ideal if you could simply import the entire device at once with the “Include Label”. Nevertheless, many thanks for the excellent work so far.

**Comment by Moonbase59** [comment]
I agree HA-Device-based Matter clusters would be great for user experience.

I reckon this must be _very carefully tested_, especially for "dumb" Matter controllers, and in any case with "the big three" (Alexa, Google Home/Assist, HomeKit).

@t0bst4r: We'll certainly _not_ hate you for such a breaking change, it's just how things go with "alpha" versions. We _do_ appreciate your hard work (and that of the contributors and makers of underlying software). Thank you all—the HA Matter Hub is a huge step forward!

**Comment by myevit** [comment]
Agree. Devices instead of entities is the way.
